src/pal/src/include/pal/utils.h

```diff
@@ -129,5 +129,29 @@ void UTIL_SetLastErrorFromMach(kern_return_t MachReturn);
 #ifdef __cplusplus
 }
 #endif // __cplusplus
-
+class StringHolder
+   {
+       private:
+           LPSTR data;
+       public:
+        StringHolder() : data(NULL) { }
+        ~StringHolder() 
+        { if(data)  
+            PAL_free( data); 
+        }
+
+        operator LPSTR () { return data;}
+
+        StringHolder& operator= (LPSTR value) 
+        {
+            data = value;
+            return *this;
+        }
+
+        BOOL IsEmpty()
+        {
+          return data == NULL || strlen(data) == 0;
```

Comment:
I don't think valid string with zero length should return true, the purpose of this method should be to check whether the holder holds a string or not. That's what was being done at the places where you are using the holder. And then a better name for this method would be IsNull as I've suggested.